### PR TITLE
add a cookie manager

### PR DESCRIPTION
**Pull request by diracdeltas**
this allows users to view all the cookies, sort them, and delete individual ones or in bulk.

fix https://github.com/brave/browser-laptop/issues/1991

todo: add ability to edit cookies and add to global whitelist/blacklist

test plan:
1. automated tests should pass
2. browse some sites in various browsing sessions (session tab, regular tab, private tab, etc.)
3. go to about:cookies
4. you should see cookies for those sites
5. click on a cookie and right-click to delete or copy. both should work.
6. repeat step 5 but for multiple selected cookies at once
7. select some cookies, hit delete key. they should be deleted.

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Test Plan:

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: js/about/cookies.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const React = require('react')
+const SortableTable = require('../../app/renderer/components/common/sortableTable')
+const aboutActions = require('./aboutActions')
+const messages = require('../constants/messages')
+const moment = require('moment')
```
Comment (diracdeltas) [inline js/about/cookies.js]: if #11061 is merged before this, i'll replace moment with date-fns

**Comment by luixxiul** [comment]
I tried to polish the design of the manager with this commit : https://github.com/brave/browser-laptop/commit/49b36b51e4a352c161fe1381ea960a945b167d0b. It would be appreciated if you would refer it. Thanks!

**Comment by diracdeltas** [comment]
^ cherry-picked

**Comment by diracdeltas** [comment]
rebased

**Comment by cezaraugusto** [comment]
looks great but after playing around for a while the browser has frozen for several times due to many cookies I received. Site visited was wired.com, quora.com, and google.com.

This is not really related to this PR but more like with the way we handle large amounts of tabular data. I created #11551 to improve this and set to the same milestone.

This is a blocker for me but a quick solution would be to limit the max number of cookies, similar to how we do for history. Not sure about the implications on privacy. 

If limiting is good enough lmk I'll make another review or otherwise, I'm working on a solution.

**Comment by luixxiul** [comment]
cc @cezaraugusto on triage (do you think this PR should be postponed?)

**Comment by diracdeltas** [comment]
marking as blocked on  #11551 for now

**Comment by alexwykoff** [comment]
this needs a design review because cookie lists can get hella long

**Comment by diracdeltas** [comment]
we're probably just going to close this in favor of waiting til chromium fork

**Comment by bsclifton** [comment]
Closing as this is marked with `post-v1` and the accompanying issue (https://github.com/brave/browser-laptop/issues/1991) has been closed

**Pull request closed without merging**